test/e2e/encryption_at_rest_test.go

```diff
@@ -64,7 +74,13 @@ var _ = Describe("Encryption at REST test", Label("encryption-at-rest"), func()
 		func(test *model.TestDataProvider, encAtRest v1.EncryptionAtRest, roles []cloudaccess.Role) {
 			testData = test
 			actions.ProjectCreationFlow(test)
-			encryptionAtRestFlow(test, encAtRest, roles)
+
+			if roles != nil {
+				cloudAccessRolesFlow(test, roles)
+				encAtRest.AwsKms.RoleID = test.Project.Status.CloudProviderAccessRoles[0].IamAssumedRoleArn
+			}
```

Comment:
Is this the fix for encryption at rest flaky tests?

Reply:
This was the actual bug. For this test case, roleID was not being set and the operator was failing to create it.
Despite there being a piece of code in the operator to handle this, it depends on having the latest version of the resource with updated status. Locally this was easily achieved and the test was always passing while in the CI we always failing.

Reply:
not always failing, but almost always right? seemed we got lucky sometimes.